### PR TITLE
Reduce redundancy while constructing `allSymbols`

### PR DESCRIPTION
**Pull request by StevenSorial**
This removes redundant symbols for `allSymbols` to reduce code/package size.

### CODE REVIEW

**Comment by fredpi** [comment]
@StevenMagdy Thanks for the PR!

Two quick thoughts:

- While the "calculation" of the `allSymbols` set takes place only once per runtime, I'd still be interested in how long it takes compared to the previous variant. Maybe you can measure it?
- This implementation currently doesn't take care of removed symbols. This is e. g. the case with deprecated symbols. Without removing them, we get duplicate symbols and issues like #97.

**Comment by StevenSorial** [comment]
> This implementation currently doesn't take care of removed symbols.

You are totally right. I completely forgot the whole reason of the current approach. 🤦🏼‍♂️ 

Closing.

**Comment by knothed** [comment]
There is no need to close this. Just generate an extra array of removed symbols for every version which are then removed from the created allSymbols set.

**Comment by StevenSorial** [comment]
@knothed Good idea. I wonder if that is going to be less code than declaring the whole Set as we currently do.

**Comment by knothed** [comment]
I believe so as there are not _that_ many deprecated symbols compared with how many there are in total.

**Comment by StevenSorial** [comment]
Removed deprecated symbols by declaring another set for each availability as suggested by @knothed 

@fredpi I did some quick measuring, and it did increase from 2ms to 3ms in an unoptimized build. I think that is ok considering that we decreased the size of the code, and it's only calculated once anyway.

**Review by knothed — APPROVED** [review]

**Comment by knothed** [comment]
Great, looks good!

**Comment by StevenSorial** [comment]
@fredpi Are you ok with merging this PR? I will rebase #102 after merging

File: Sources/SFSafeSymbols/Symbols/SFSymbol+AllSymbols.swift [hunk 1]
```diff
@@ -6,20 +6,31 @@ extension SFSymbol {
     public static var allCases: [SFSymbol] { Array(allSymbols) }
 
     public static var allSymbols: Set<SFSymbol> = {
+        var result = allSymbols1_0
+        if #available(iOS 14.0, tvOS 14.0, watchOS 7.0, *) {
+            result.formUnion(allSymbols2_0)
```
Comment (fredpi) [inline Sources/SFSafeSymbols/Symbols/SFSymbol+AllSymbols.swift]: [Optional] I would suggest to use another naming: `symbolsAvailableSince2_0`. It's an `internal` variable and one could argue that naming is not so important, but `allSymbols2_0` is not the proper description of what's behind the variable.

**Review by fredpi — APPROVED** [review]
@StevenMagdy Thank you very much! I just proposed one optional improvement; you can decide by yourself whether you want to implement it or not and then merge this PR.

File: Sources/SFSafeSymbols/Symbols/SFSymbol+AllSymbols+1.0.swift [hunk 2]
```diff
@@ -1647,7 +1647,7 @@ extension SFSymbol {
             zCircleFill,
             zSquare,
             zSquareFill,
-            zzz
+            zzz,
```
Comment (fredpi) [inline Sources/SFSafeSymbols/Symbols/SFSymbol+AllSymbols+1.0.swift]: Ah, and one other little detail: I would suggest not to put a comma behind the last entry of the set

**Comment by StevenSorial** [comment]
Feedback implemented. Thanks for the review.

**Pull request merged**